### PR TITLE
Feature: Keybinds for user specified commands

### PR DESCRIPTION
**Pull request by kristofbolyai**
Adds the ability to set 6 keybinds for 6 different command, changable in Wynntils configs.

This pull request is an improvement of #402.

Some texts might require rewording.

### CODE REVIEW

File: src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java [hunk 1]
```diff
@@ -91,6 +92,72 @@ public static void registerKeys() {
         CoreModule.getModule().registerKeyBinding("View Player's Gear", -98, "Wynntils", KeyConflictContext.IN_GAME, true, GearViewerUI::openGearViewer);
 
         showLevelOverlayKey = UtilitiesModule.getModule().registerKeyBinding("Show Item Level Overlay", Keyboard.KEY_LCONTROL, "Wynntils", WynntilsConflictContext.AMBIENT, true, () -> {});
+
+        CoreModule.getModule().registerKeyBinding("Command keybind 1", Keyboard.KEY_NONE, "Wynntils", KeyConflictContext.IN_GAME, true, () -> {
```
Comment (Incompleteusern) [inline src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java]: The keybind part is repeated for every single keybind, probably separate it with a function

File: src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java [hunk 2]
```diff
@@ -91,6 +92,72 @@ public static void registerKeys() {
...
+            String cKeyBind = UtilitiesConfig.CommandKeybinds.INSTANCE.cKeyBind1;
+            if (cKeyBind.isEmpty())
+                return;
+
+            if (cKeyBind.startsWith("/"))
+                McIf.player().sendChatMessage(cKeyBind);
+            else
+                McIf.player().sendChatMessage("/" + cKeyBind);
```
Comment (Incompleteusern) [inline src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java]: This should be done outside the keybind itself, perhaps in `public void onSettingChanged(String name) {` of line 503 in `UtilitiesConfig`

Comment (Incompleteusern) [reply src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java]: Also, add a check for if the player is in a gui before running the command

**Review by Incompleteusern — COMMENTED** [review]
LGTM, although the keybinds don't seem to be implemented yet. Also a few things

**Comment by kristofbolyai** [comment]
> LGTM, although the keybinds don't seem to be implemented yet.

Wdym, by that? Keybinds shouldn't be set to any default IMO, and should be decided by the user.

**Comment by Incompleteusern** [comment]
oh Im dumb

File: src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java (hunk 2 above)
Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/managers/KeyManager.java]: > Also, add a check for if the player is in a gui before running the command

Seemingly no need to do that as the keybinds dont work with open guis. Am I missing something?

**Comment by kristofbolyai** [comment]
Fixed the issues, also minor change would just prefixing "/" at runtime be okay, its much cleaner and in the end it does not allow spamming chat messages so it should be alright?

**Review by magicus — APPROVED** [review]
This sounds like a reasonable functionality to add.

As a separate follow-up, we might consider giving hints on suitable commands, like how the %info% variables work.

**Pull request merged**